### PR TITLE
limit description length

### PR DESCRIPTION
**Pull request by chaol12**

### CODE REVIEW

File: src/pages/AddTransactionForm.jsx [hunk 1]
```diff
@@ -41,7 +41,7 @@ export default function AddTransactionForm({fetchTransactions}) {
           addTransactionToDb(newTransaction) 
         }
         else if (today < transactionDate){
-          alert("Never know tomorrow ")
+          alert("That is in the future")
```
Comment (aappacode) [inline src/pages/AddTransactionForm.jsx]: Maybe we can update this to be: "The transaction date you entered is in the future. Please update the date to be today's date or before today."

Comment (aappacode) [inline src/pages/AddTransactionForm.jsx]: Also, how far back can they update something? If they are adding something for the previous month, do our reports change/have to be regenerated?

File: src/pages/AddTransactionForm.jsx [hunk 2]
```diff
@@ -61,12 +61,18 @@ export default function AddTransactionForm({fetchTransactions}) {
 
     const handleSubmit = () => {
       setOpen(false);
-      if(newAmount){
+      if(newAmount && newDescription.length < 50 && newDescription.length > 0 ){
         const absAmount = Math.abs(parseFloat(newAmount).toFixed(2))
         addTransactions(newDescription,absAmount,transactionDate)
+      }else if(newAmount && (newDescription.length > 50 || newDescription.length === 0)){
+        alert("fail to add new transaction, the description is too long or empty")
```
Comment (aappacode) [inline src/pages/AddTransactionForm.jsx]: Update alert text to "Failed to add a new transaction, the description you entered is too long or empty"

Or you can separate the length conditional and empty conditional to give more precise alerts. Up to you!

For example (pseudocode):
`if(new.Description.length==0) { alert("Failed to add a new transaction, the description you entered is empty")} else if(new.Description.length < 50 ) { alert("Failed to add a new transaction, the description you entered is too long")}`

File: src/pages/AddTransactionForm.jsx [hunk 3]
```diff
@@ -61,12 +61,18 @@ export default function AddTransactionForm({fetchTransactions}) {
...
+        setNewDescription("")
+        setAmount("")
       }else
       {
-        alert("fail to add new transaction, you didn't enter the new amount correctly")
+        alert("fail to add new transaction, you didn't enter the amount correctly")
```
Comment (aappacode) [inline src/pages/AddTransactionForm.jsx]: Update text to "Failed to add the new transaction, the amount you entered was incorrectly entered. Please x" (provide them what they should be entering instead/example.

**Review by aappacode — COMMENTED** [review]
Review comments on alert comment/texts

File: src/pages/AddTransactionForm.jsx (hunk 1 above)
Comment (chaol12) [reply src/pages/AddTransactionForm.jsx]: I think it's fine to add something to the past, but I think I should put restrictions on it, so I limit it within a year. for example,  we could add 2022 - today's transactions

**Review by aappacode — APPROVED** [review]
Thanks for updating the alert messages!

**Pull request merged**